src/VisualStudio/Core/Impl/CodeModel/CodeModelTaskExtensions.cs

```diff
@@ -16,13 +14,12 @@ internal static class CodeModelTaskExtensions
         /// Does a Roslyn.Utilities.TaskExtensions.WaitAndGetResult{T} for CodeModel.
         /// </summary>
         /// <remarks>
-        /// This function is the exact same as Roslyn.Utilities.TaskExtensions.WaitAndGetResult{T}, except it opts out
-        /// of enforcement that it can be called on non-UI threads. CodeModel, since it must implement a highly blocking API,
-        /// has no choice but to use WaitAndGetResult in a bunch of places. But that's not a good reason to require the tests
-        /// to have thread affinity, since the tests have no specific threading requirements. Thus, it's acceptable for CodeModel
-        /// to call the _CanCallOnBackground variant. We hope to audit _CanCallOnBackground periodically, and so rather than
-        /// having to understand that each of those uses are CodeModel and thus get a special pass.</remarks>
+        /// This function is the exact same as Roslyn.Utilities.TaskExtensions.WaitAndGetResult{T}. CodeModel, since it
+        /// must implement a highly blocking API, has no choice but to use WaitAndGetResult in a bunch of places. We
+        /// hope to audit _CanCallOnBackground periodically, and so rather than having to understand that each of those
+        /// uses are CodeModel and thus get a special pass.
+        /// </remarks>
         public static T WaitAndGetResult_CodeModel<T>(this Task<T> task, CancellationToken cancellationToken)
-            => task.WaitAndGetResult_CanCallOnBackground(cancellationToken);
+            => task.WaitAndGetResult(cancellationToken);
```

Comment:
I'll admit this worries me a bit only because I can't explain why _CanCallOnBackground was being used in the first place. I guess we'll find out.

Reply:
Prior to 2d6d3199, the tests were not using the main thread.